WcaOnRails/app/controllers/api/v0/roles_controller.rb

```diff
@@ -107,21 +118,67 @@ def index_for_group_type
     group_ids = UserGroup.where(group_type: group_type).pluck(:id)
     roles = Role.where(group_id: group_ids).to_a # to_a is for the same reason as in index_for_user.
 
-    # Temporary hack to support the old delegate structure, will be removed once all roles are
+    # Temporary hack to support the old system roles, will be removed once all roles are
     # migrated to the new system.
     if group_type == UserGroup.group_types[:delegate_regions]
       roles.concat(User.where.not(delegate_status: nil).map(&:delegate_role))
+    elsif group_type == UserGroup.group_types[:councils]
+      Team.all_councils.each do |council|
+        leader = TeamMember.find_by(team_id: council.id, team_leader: true, end_date: nil)
+        if leader.present?
+          roles << {
+            id: group_type + "_" + leader.id.to_s,
+            group: {
+              id: group_type + "_" + council.id.to_s,
+              name: council.name,
+              group_type: UserGroup.group_types[:councils],
+              is_hidden: false,
+              is_active: true,
+            },
+            user: leader.user,
+            metadata: {
+              status: 'leader',
+            },
+          }
+        end
+      end
     end
 
     # Filter the list based on the permissions of the logged in user.
     roles = filter_roles_for_logged_in_user(roles)
 
     # Filter the list based on the other parameters.
-    roles = filter_roles_for_parameters(roles, params[:status])
+    roles = filter_roles_for_parameters(roles, params[:status], params[:isActive])
 
     render json: roles
   end
 
+  def create
+    user_id = params.require(:userId)
+    group_id = params.require(:groupId)
+
+    if group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
+      # migrated to the new system.
+      original_group_id = group_id.split("_").last
+      group_type = group_id.split("_").first
```

Comment:
You're listing the line that reads `.last` first and the line that reads `.first` last. This is a bit confusing in terms of "code aesthetics" :)